packages/frontend/src/components/create-campaign/preview/index.tsx

```diff
@@ -144,11 +153,75 @@ export function CampaignPreview({
         if (payload.kpiSpecification && rewardsApproved) uploadSpecification();
     }, [payload.kpiSpecification, rewardsApproved]);
 
+    const handleSafeTransaction = useCallback((tx: BaseTransaction) => {
+        setSafeTxs((txs) => {
+            txs.push(tx);
```

Comment:
We should avoid mutating the state directly, we can use spread operator here to add the new tx.

Reply:
Addressed in 849c227d6bcc5b773cbffca69dce9732881cbb51